collect_app/src/main/java/org/odk/collect/android/tasks/FormLoaderTask.java

```diff
@@ -317,7 +317,7 @@ private boolean initializeForm(FormDef formDef, FormEntryController fec) {
         boolean usedSavepoint = false;
         if (instancePath != null) {
             File instance = new File(instancePath);
-            final File shadowInstance = SaveToDiskTask.savepointFile(instance);
+            final File shadowInstance = SaveToDiskTask.getSavepointFile(instance.getName());
```

Comment:
This could become a `SaveToDiskTask.doesSavepointFileExistForName()` method.

Reply:
Why? We want to get a file here.

Reply:
Ah, I see now. The logic here is a bit confusing. Create a new File, get a savepoint file based on its name, then if it exists ignore the old file and use the savepoint "shadow" one (what is a shadow instance?). Can you add some comments explaining why we're overwriting the instance variable and what that means?